HW-12: spring security form auth and jwt for rest controllers

OksiBlack opened the pull request:
HW-12: spring security form auth and jwt for rest controllers

Commit:
HW-11: Add keycloak config.

Commit:
HW-12: Add spring security form login.

Commit:
HW-12: Keycloak for rest controllers.

Commit:
HW-12: Formatting and test fixes.

Commit:
HW-12: Move to profile.

vlad-iv commented on hw12-spring-security/src/test/java/ru/otus/hw/controller/BookControllerTest.java:
```diff
@@ -0,0 +1,244 @@
+
+    @MockitoBean
+    private BookToSaveRequestConverter bookToSaveRequestConverter;
+
```
Предлагаю добавить тесты, если пользователь не авторизован
что на любой запрос книги будет редирект на логин

Commit:
HW-12: Tweak pipe.

Commit:
HW-12: Clean.

Commit:
HW-12: Fix clients scopes. Tweak SpringDocConfig. Tweak security.

- Tweak SpringDocConfig.
-  Clean.
- Tweak security.
- Config tweak.
- Add more method security.

Commit:
HW-12: Add security tests.

Commit:
HW-17: Clean parent pom.

Commit:
HW-17: Fix SpringDocConfig.
